### ISSUE TITLE
reserved plugin names

### ISSUE THREAD

**Issue by amoore877** [spiffe/spire#1052]
Minimal effort to protect external plugin developers from themselves. On loading plugins, do not allow (or WARN log against) plugins of type/name:
- `join_token`, as this is actually used in logical branching on attestation
- some existing built-in plugin
-- unless we also are explicitly documenting, **and logging**, the override in catalog plugin map, **and** this override happens consistently for all plugins (either built-in is always overridden or external is always overridden)

Whether or not the behavior is allowed, should at least still check **and log** when such an occurrence happens on filling the catalog. Also update documentation to state the behavior and reserved/in-use names.

**Comment by amoore877** [comment]
Ensure https://github.com/spiffe/spire/blob/master/doc/spire_server.md is up-to-date; make sure contributing doc mentions keeping it up to date with the list of plugins

**Comment by amoore877** [comment]
@azdagron with the ~1.0 refactoring of catalog architecture, is this still a valid issue? Can an external plugin take the name of an built-in one?

**Comment by azdagron** [comment]
External plugins can still take the name of a built-in one and are loaded in lieu of built-ins (by design). The plugin loading log message includes whether or not the plugin is external or not.

The attestation logic still branches on `join_token` and will not invoke an external plugin of the same name. I agree this isn't the best experience and we should probably at least be warning on this particular setup.

We can probably still use some documentation to clarify these behaviors.

In short, yes, this is still a valid issue.
